pandora-client-web/src/components/settings/securitySettings.tsx

```diff
@@ -0,0 +1,273 @@
+			</div>
+		</FieldsetToggle>
+	);
+}
+
+function ConnectedClientsList({ connections }: { connections: AccountConnectedClients | null; }): ReactElement {
+	if (connections === null)
+		return <>Loading...</>;
+
+	return (
+		<table>
+			<thead>
+				<tr>
+					<th>Id</th>
+					<th>Connections</th>
+					<th>Characters</th>
+					<th>Actions</th>
+				</tr>
+			</thead>
+			<tbody>
+				{
+					connections.map((c) => <ConnectedClientConnection key={ c.loginTokenId } connection={ c } />)
+				}
+			</tbody>
+		</table>
+	);
+}
+
+function ConnectedClientConnection({ connection }: { connection: AccountConnectedClients[number]; }): ReactElement {
+	const directoryConnector = useDirectoryConnector();
+	const authToken = useObservable(directoryConnector.authToken);
+	const confirm = useConfirmDialog();
+
+	const { loginTokenId, connectionCount, connectedCharacters } = connection;
+
+	const disconnect = React.useCallback(() => {
+		void confirm('Are you sure you want to disconnect this client?').then((confirmed) => {
+			if (confirmed) {
+				directoryConnector.sendMessage('logout', { type: 'selected', accountTokenId: loginTokenId });
+			}
+		});
+	}, [confirm, directoryConnector, loginTokenId]);
+
+	const isCurrent = authToken?.value.startsWith(loginTokenId);
+
+	return (
+		<tr className={ isCurrent ? 'current-connection' : undefined }>
+			<td>{ loginTokenId }</td>
+			<td>{ connectionCount }</td>
+			<td>{ connectedCharacters.map(({ id, name }) => `${name} (${id})`).join(', ') }</td>
+			<td>
+				{
+					isCurrent ? (
+						<>
+							<Button className='slim' onClick={ disconnect }>Logout</Button>
+							<ExtendCurrentSession token={ authToken! } />
+						</>
+					) : (
+						<Button className='slim' onClick={ disconnect }>Disconnect</Button>
+					)
+				}
+			</td>
+		</tr>
+	);
+}
+
+function ExtendCurrentSession({ token }: { token: AuthToken; }): ReactElement {
+	const [show, setShow] = React.useState(false);
+
+	const hide = React.useCallback(() => {
+		setShow(false);
+		return true;
+	}, []);
+
+	useKeyDownEvent(hide, 'Escape');
+
+	if (!show)
+		return <Button className='slim' onClick={ () => setShow(true) }>Extend</Button>;
+
+	return (
+		<>
+			<Button className='slim' onClick={ hide }>Extend</Button>
+			<ExtendCurrentSessionDialog token={ token } hide={ hide } />
+		</>
+	);
+}
+
+function ExtendCurrentSessionDialog({ token, hide }: { token: AuthToken; hide: () => boolean; }): ReactElement {
+	const directory = useDirectoryConnector();
+	const [password, setPassword] = React.useState('');
+	const now = useCurrentTime(60_000);
+
+	const [extend, processing] = useAsyncEvent(
+		() => directory.extendAuthToken(password),
+		(result) => {
+			if (result)
+				hide();
+			else
+				setPassword('');
+		},
+	);
+
+	return (
+		<ModalDialog>
+			<Form dirty={ false } onSubmit={ extend }>
+				<p>Your session will expire in { FormatTimeInterval(token.expires - now, 'short') }</p>
```

Comment:
Is short time the best format here? Is there any need to keep it limited to the highest unit of time? Short format is primarily meant for places that show quick overview of many items, each having some time, but details should always show most detailed time available.
Also as I said higher I didn't even notice this was here until reading through the code. And the extend password popup is most definitely not a place I would look for this information in.